<launch>
    <arg name="veh" doc="Name of vehicle. ex: megaman"/>
    <arg name="config" default="baseline" doc="Specify a config."/>
    <arg name="param_file_name" default="default" doc="Specify a param file. ex:megaman. Note that you will need a megaman.yaml in the config folder for every node you are launching. To test with only one megaman.yaml, the recommendation is to set this to default and then manually change the desired param file in the approprate rosparam tag below"/>

    <arg name="test" default="false"/>

    <!-- Vehicle Coordinator -->
    <include file="$(find veh_coordinator)/launch/simple_coordinator_node.launch">
      <arg name="veh" value="$(arg veh)"/>
      <arg name="config" value="$(arg config)"/>
      <arg name="param_file_name" value="$(arg param_file_name)"/>
      <arg name="intersectionType" value="trafficLight"/>
    </include>

    <remap from="fake_duckiebot_node/mode" to="simple_coordinator_node/mode" />
    <remap from="fake_duckiebot_node/signals_detection" to="simple_coordinator_node/signals_detection" />
    <remap from="fake_duckiebot_node/change_color_pattern" to="simple_coordinator_node/change_color_pattern" />
    <remap from="fake_duckiebot_node/clearance_to_go" to="simple_coordinator_node/clearance_to_go"  />

    <!-- Fake inputs for Vehicle Coordinator -->
    <include file="$(find veh_coordinator)/launch/fake_duckiebot_node.launch">
      <arg name="veh" value="$(arg veh)"/>
      <arg name="config" value="$(arg config)"/>
      <arg name="param_file_name" value="$(arg param_file_name)"/>
    </include>
</launch>